components/region_cache_memory_engine/src/write_batch.rs

```diff
@@ -279,8 +318,13 @@ impl Mutable for RangeCacheWriteBatch {
     }
 
     fn put_cf(&mut self, cf: &str, key: &[u8], val: &[u8]) -> Result<()> {
-        self.buffer
-            .push(RangeCacheWriteBatchEntry::put_value(cf, key, val));
+        if self.range_in_cache {
+            self.buffer
+                .push(RangeCacheWriteBatchEntry::put_value(cf, key, val));
+        } else if self.pending_range_in_loading {
+            self.pending_range_in_loading_buffer
+                .push(RangeCacheWriteBatchEntry::put_value(cf, key, val));
+        }
```

Comment:
Can the code of put_cf and delete_cf be shared on this code path? 
This is critical part that missing any entry could lead to data loss.

Reply:
Done